apps/marketplace-controller/contracts/MarketplaceController.sol

```diff
@@ -167,6 +169,19 @@ contract MarketplaceController is EtherTokenConstant, IsContract, IMarketplaceCo
         marketMaker.makeSellOrder(msg.sender, _collateral, _sellAmount, _minReturnAmountAfterFee);
     }
 
+    /**
+     * @dev ApproveAndCallFallBack interface conformance
+     * @param _from Token sender
+     * @param _amount Token amount
+     * @param _token Token that received approval
+     * @param _buyOrderData Data for the below function call
+     *      makeBuyOrder(address _buyer, address _collateral, uint256 _depositAmount, uint256 _minReturnAmountAfterFee)
+    */
+    function receiveApproval(address _from, uint256 _amount, address _token, bytes _buyOrderData) public {
+        require(canPerform(_from, MAKE_BUY_ORDER_ROLE, new uint256[](0)), ERROR_NO_PERMISSION);
+        marketMaker.makeBuyOrderRaw(_from, msg.sender, _amount, _buyOrderData);
```

Comment:
I thought about that and decided I didn't think it was necessary. If somehow a malicious collateral token was used, I can't see how it specifying a different `_token` when calling this function could cause any issues since it's not used.

Reply:
Hmm good point,  can't think on any specific issue either. 

Was thinking it more in the context that we can know that `_token` is the address of the trusted token. But in any case, malicious tokens could do that too so i think is fine as it is.